CHANGELOG.md

```diff
@@ -1,5 +1,7 @@
 ## Unreleased
 
+# 1.4.1 (October 26th, 2023)
+
 CHANGES:
 
 * Build with Go v1.21.3
```

Comment:
Also looks like we need to add `sigs.k8s.io/secrets-store-csi-driver v1.3.2 -> v1.3.3` to the list of updated deps